code/go/0chain.net/smartcontract/storagesc/challengepool.go

```diff
@@ -364,15 +365,24 @@ func (ssc *StorageSmartContract) getChallengePoolStatHandler(
 	)
 
 	if allocationID == "" {
-		return nil, errors.New("missing allocation_id URL query parameter")
+		err := errors.New("missing allocation_id URL query parameter")
+		return nil, smartcontract.WrapErrInvalidRequest(err)
 	}
 
 	if alloc, err = ssc.getAllocation(allocationID, balances); err != nil {
-		return
+		intErr := smartcontract.NewError(smartcontract.FailRetrievingAllocationErr, err)
+
+		switch {
+		case errors.Is(err, util.ErrValueNotPresent):
+			return nil, smartcontract.WrapErrNoResource(intErr)
+		default:
+			return nil, smartcontract.WrapErrInternal(intErr)
+		}
```

Comment:
if errors.Is(err, util.ErrValueNotPresent) {
    return nil, smartcontract.WrapErrNoResource(intErr)
}
return nil, smartcontract.WrapErrInternal(intErr)

Just for consistency across different parts of the application.

Reply:
I added the `NewErrNoResourceOrErrInternal` method solving the boilerplate code problem.